Engine/Generic/RuleSuppression.cs

```diff
@@ -49,28 +48,8 @@ public int EndOffset
         /// </summary>
         public string RuleName
         {
-            get
-            {
-                return _ruleName;
-            }
-
-            set
-            {
-                _ruleName = value;
-
-                if (!String.IsNullOrWhiteSpace(_ruleName)
-                    && (ScriptAnalyzer.Instance.ScriptRules != null
-                        && ScriptAnalyzer.Instance.ScriptRules.Count(item => String.Equals(item.GetName(), _ruleName, StringComparison.OrdinalIgnoreCase)) == 0)
-                    && (ScriptAnalyzer.Instance.TokenRules != null
-                        && ScriptAnalyzer.Instance.TokenRules.Count(item => String.Equals(item.GetName(), _ruleName, StringComparison.OrdinalIgnoreCase)) == 0)
-                    && (ScriptAnalyzer.Instance.ExternalRules != null
-                        && ScriptAnalyzer.Instance.ExternalRules.Count(item => String.Equals(item.GetFullName(), _ruleName, StringComparison.OrdinalIgnoreCase)) == 0)
-                    && (ScriptAnalyzer.Instance.DSCResourceRules != null
-                        && ScriptAnalyzer.Instance.DSCResourceRules.Count(item => String.Equals(item.GetName(), _ruleName, StringComparison.OrdinalIgnoreCase)) == 0))
```

Comment:
As one can see in the combination of all those statements is that we do not know whether the rule name to be  suppressed is of type custom rule (where the rule name cannot be determined at design time) or not. Therefore it is not possible to make a statement whether the given rule name will be in one of the returned DiagnosticRecords